gnosis/cowswap
# [presign] Get Gnosis Safe transaction state and update ethereumTransactionHash

# Summary

This PR includes some new logic in the `FinalizeTxUpdater` logic to:
* 1. Periodically check the state of the gnosis safe transaction
* 2. Update the state with the safe info
* 3. Update the txHash if available
* 4. Obtains the receipt of the Ethereum transaction when it's available
* 5. Finalize the transaction, the same way normal transactions are finalized


![image](https://user-images.githubusercontent.com/2352112/134546450-b0aec8b1-4c74-4c4c-ab91-5e28ab638ea2.png)

# Not included
The UX is still not perfect. There's a few details missing, for example if you click in the transaction shown in the side panel, it will take you to a non existing transaction. This is because, for now it just shows you the Gnosis Safe tx hash. I plan to address this in the future. @biocom This relates to create the 2 links: safe tx, etherscan

# To Test
This PR should make approvals/wraps/unwraps work with Gnosis Safe.

You might experience a small delay from when the transaction is mined and when it shows as finalized in CowSwap, however this delay should be a few seconds at most.

For dev testers, you can check the REDUX actions, how periodiacally is updating with the latest Gnosis Safe Tx info, until we have a mined transaction, then we would invoke the tx finalization.
